cypress/e2e/audio-player/audio-player.spec.ts

```diff
@@ -0,0 +1,588 @@
+
+                cy.get(".mx_EventTile_last.mx_EventTile_selected").within(() => {
+                    checkPlayerVisibility();
+                });
+
+                cy.get(".mx_EventTile_last").percySnapshotElement(
+                    `Selected EventTile of audio player with a reply on ${layout} layout`,
+                    {
+                        percyCSS,
+                        widths: snapshotWidthsEventTile,
+                    },
+                );
+                cy.log(`Took a snapshot of selected EventTile of audio player with a reply on ${layout} layout`);
+            };
+
+            // Take a snapshot of EventTile with a reply on group layout
+            cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.Group);
+            takeSnapshotReply("group");
+
+            // Take a snapshot of EventTile with a reply on bubble layout
+            cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.Bubble);
+            takeSnapshotReply("bubble");
+        });
+    });
+
+    it("should support creating a reply chain with multiple audio files", () => {
+        // Note: "mx_ReplyChain" element is used not only for replies which
+        // create a reply chain, but also for a single reply without a replied
+        // message. This test checks whether a reply chain which consists of
+        // multiple audio file replies is rendered properly.
+
+        const clickButtonReply = () => {
+            cy.get(".mx_EventTile_last")
+                .realHover()
+                .within(() => {
+                    // Click "Reply" button on MessageActionBar
+                    cy.get('[aria-label="Reply"]').click({ force: false });
+                });
+        };
+
+        visitRoom();
+
+        // Upload one second audio file with a long file name
+        uploadFile("cypress/fixtures/1sec-long-name-audio-file.ogg");
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            // Assert the audio player is rendered
+            cy.get(".mx_EventTile_last .mx_AudioPlayer_container").should("exist");
+
+            clickButtonReply();
+        });
+
+        // Reply to the player with another audio file
+        uploadFile("cypress/fixtures/1sec.ogg");
+
+        cy.get(".mx_RoomView_MessageList").within(() => {
+            cy.get(".mx_EventTile_last").within(() => {
+                // Assert the audio player is not rendered
+                cy.get(".mx_EventTile_last .mx_AudioPlayer_container").should("not.exist");
+
+                // Assert that audio file is rendered as file button
+                cy.get(".mx_ReplyChain").within(() => {
+                    cy.get(".mx_MFileBody_info[role='button']").within(() => {
+                        // Assert that the file button has file name
+                        cy.get(".mx_MFileBody_info_filename").should("exist");
+                    });
+                });
+            });
```

Comment:
it seems like this is repeating checks from the previous test ("should support replying to audio file with another audio file") - can we omit it here?

Reply:
Sure, we should not repeat the same thing here, which indeed will make it difficult to debug once an error happens due to ... a reply or a reply chain.